elixir/apps/domain/lib/domain/directories/provider.ex

```diff
@@ -0,0 +1,24 @@
+defmodule Domain.Directories.Provider do
+  use Domain, :schema
+
+  @types ~w[
+    okta
+  ]a
+
+  schema "directory_providers" do
+    belongs_to :account, Domain.Accounts.Account
+    belongs_to :auth_provider, Domain.Auth.Provider
```

Comment:
Just commenting here to reference my comment above about not need to have an `auth_provider` reference in a `directory_provider`.

Reply:
Circling back on this, I think having this here is going to be the cleanest stepping stone for now, and then we can remove it when we have the new credentials flow set up.

The reason is because currently identities and groups are tied to the auth_provider, and it's not straightforward to just replace that foreign_key with directory_providers_id because it could also be linked to non-syncing providers like OIDC and Email.

For the short-term it's feeling cleaner for `auth_provider` to have one `directory_provider` whenever that auth provider has sync enabled. The sync fields will be moving to `directory_providers` and we'll enforce each auth_provider can only have one directory_provider (or none in the case of OIDC/Email).

This is all non-breaking for now.

Reply:
auth_provider is also looked up when the user authenticates, so we need to keep the relationship between identity and auth_provider for the time being. Can chip away at improving things here.